src/common/BackTrace.cc

```diff
@@ -2,29 +2,33 @@
 // vim: ts=8 sw=2 smarttab
 
 #include <ostream>
-#include <cxxabi.h>
 #include <string.h>
 
+#include <boost/stacktrace.hpp>
```

Comment:
i'd suggest move the `crommon/BackTrace.cc` change out into a separated PR, so we can have this change merged sooner. also, AFAIR, the output format of demangled symbol of boost.backtrace is a little bit different from what we have now. not sure if this matters.

Reply:
```
[ RUN      ] BackTrace.Basic
../src/test/common/test_back_trace.cc:43: Failure
Value of: std::regex_match(lines[lineno], e)
  Actual: false
Expected: true
[  FAILED  ] BackTrace.Basic (2 ms)
```

indeed.

Reply:
the format does matter for crash-reports that telemetry analyses - @yaarith fyi

Reply:
Sure, dissected into https://github.com/ceph/ceph/pull/40947.